packages/components/src/number-control/test/index.tsx

```diff
@@ -0,0 +1,616 @@
+/**
+ * External dependencies
+ */
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import NumberControl from '..';
+import type { NumberControlProps } from '../types';
+
+const getInput = () => screen.getByRole( 'spinbutton' ) as HTMLInputElement;
+
+function StatefulNumberControl( props: NumberControlProps ) {
+	const [ value, setValue ] = useState( props.value );
+	const handleOnChange = ( v: string | undefined ) => setValue( v );
+
+	return (
+		<NumberControl
+			{ ...props }
+			value={ value }
+			onChange={ handleOnChange }
+		/>
+	);
+}
+
+describe( 'NumberControl', () => {
+	describe( 'Basic rendering', () => {
+		it( 'should render', () => {
+			render( <NumberControl /> );
+			expect( getInput() ).toBeInTheDocument();
```

Comment:
In my experience so far I've found `.toBeVisible()` to be even more specific than `.toBeInTheDocument()`. It often is what we want to ensure, too - that the user can see and experience it.

```suggestion
			expect( getInput() ).toBeVisible();
```

Reply:
Makes sense, I applied those changes